components/canvas/webgl_thread.rs

```diff
@@ -1233,47 +1272,72 @@ impl WebGLImpl {
+                sender
+                    .send(value.map(|ntf| ntf.0.get()).unwrap_or_default())
+                    .unwrap()
             },
             WebGLCommand::DeleteTransformFeedback(id) => {
-                gl.delete_transform_feedbacks(id);
+                if let Some(tf) = NonZeroU32::new(id) {
+                    unsafe { gl.delete_transform_feedback(NativeTransformFeedback(tf)) };
+                }
             },
             WebGLCommand::IsTransformFeedback(id, ref sender) => {
-                let value = gl.is_transform_feedback(id);
+                let value = NonZeroU32::new(id)
+                    .map(|id| unsafe { gl.is_transform_feedback(NativeTransformFeedback(id)) })
+                    .unwrap_or_default();
                 sender.send(value).unwrap()
             },
             WebGLCommand::BindTransformFeedback(target, id) => {
-                gl.bind_transform_feedback(target, id);
+                unsafe {
+                    gl.bind_transform_feedback(
+                        target,
+                        NonZeroU32::new(id).map(NativeTransformFeedback),
+                    )
+                };
             },
             WebGLCommand::BeginTransformFeedback(mode) => {
-                gl.begin_transform_feedback(mode);
+                unsafe { gl.begin_transform_feedback(mode) };
             },
             WebGLCommand::EndTransformFeedback() => {
-                gl.end_transform_feedback();
+                unsafe { gl.end_transform_feedback() };
             },
             WebGLCommand::PauseTransformFeedback() => {
-                gl.pause_transform_feedback();
+                unsafe { gl.pause_transform_feedback() };
             },
             WebGLCommand::ResumeTransformFeedback() => {
-                gl.resume_transform_feedback();
+                unsafe { gl.resume_transform_feedback() };
             },
             WebGLCommand::GetTransformFeedbackVarying(program, index, ref sender) => {
-                let (size, ty, mut name) = gl.get_transform_feedback_varying(program.get(), index);
+                let ActiveTransformFeedback { size, tftype, name } =
+                    unsafe { gl.get_transform_feedback_varying(program.glow(), index) }.unwrap();
                 // We need to split, because the name starts with '_u' prefix.
-                name = name.split_off(2);
-                sender.send((size, ty, name)).unwrap();
+                let name = from_name_in_compiled_shader(&name);
+                sender.send((size, tftype, name)).unwrap();
             },
             WebGLCommand::TransformFeedbackVaryings(program, ref varyings, buffer_mode) => {
-                gl.transform_feedback_varyings(program.get(), varyings.as_slice(), buffer_mode);
+                let varyings: Vec<String> = varyings
+                    .iter()
+                    .map(|varying| to_name_in_compiled_shader(varying))
```

Comment:
```suggestion
                    .map(to_name_in_compiled_shader)
```

Reply:
This won't work as it will not coerce &String int &str, that's also why https://doc.rust-lang.org/std/convert/fn.identity.html exists.